src/library/rekordbox/rekordboxfeature.cpp

```diff
@@ -993,13 +1016,26 @@ void readAnalyze(TrackPointer track, double sampleRate, int timingOffset, bool i
         }
     }
 
-    if (cueLoadPosition < kLongestPosition) {
-        track->setCuePoint(CuePosition(cueLoadPosition));
+    std::sort(memoryCues.begin(), memoryCues.end(), [](const memory_cue_t& a, const memory_cue_t& b) -> bool {
+        return a.position < b.position;
+    });
+
+    if (memoryCues.size() > 0) {
+        // Set first chronological memory cue as Mixxx MainCue
+        memory_cue_t firstCue = memoryCues[0];
+        track->setCuePoint(CuePosition(firstCue.position));
         CuePointer pLoadCue = track->findCueByType(mixxx::CueType::MainCue);
-        if (!cueLoadComment.isNull()) {
-            pLoadCue->setLabel(cueLoadComment);
+        if (!firstCue.comment.isNull()) {
```

Comment:
Is this check really necessary?

Reply:
If setting a hotcue label to an empty QString doesn’t incur any extra overhead, is not needed

Reply:
I don't think there's overhead, and I worry about edge cases where you already have a main cue with a label, then import data from rekordbox where your main cue does *not* have a label and you end up with new main cue that still has the old label.